legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -309,7 +309,10 @@ else if (matchingQueries.size() == 0)
         {
             throw new ApplicationQueryException("Can't find query with ID '" + queryId + "'", Response.Status.NOT_FOUND);
         }
-        return matchingQueries.get(0);
+        Query query = matchingQueries.get(0);
+        query.lastOpenAt = Instant.now().toEpochMilli();
+        this.getQueryCollection().findOneAndReplace(Filters.eq("id", queryId), queryToDocument(query));
```

Comment:
can we not do a full find and replace here and just update the lastOpenAt property instead ?

Reply:
this getQuery()  is not created by this PR.
What I am doing here is just updating the `lastOpenAt `

Reply:
sry May I know how this new information will be stored back in Mongo if not doing this 
` this.getQueryCollection().findOneAndReplace(Filters.eq("id", queryId), queryToDocument(query));`

Reply:
couldnt we just set the lastOpenAt property, something like -> https://www.mongodb.com/docs/manual/tutorial/update-documents/#update-a-single-document. 
I think it might be safer to just update that single property because really you shouldnt be updating anything else in a get request.